src/main/java/org/alfasoftware/soapstone/SoapstoneOpenApiReader.java

```diff
@@ -413,6 +423,40 @@ private RequestBody parametersToRequestBody(String operationId, Collection<Param
     JavaType javaType = soapstoneConfiguration.getObjectMapper().constructType(type);
     LOG.debug("          " + javaType.toString());
 
+    XmlJavaTypeAdapter adapterAnnotation = currentResourceClass.getPackage().getAnnotation(XmlJavaTypeAdapter.class);
+    if (adapterAnnotation == null || adapterAnnotation.type() != type) {
+      XmlJavaTypeAdapters adaptersAnnotation = currentResourceClass.getPackage().getAnnotation(XmlJavaTypeAdapters.class);
+
+      final Type finalType = type;
+      adapterAnnotation = adaptersAnnotation == null ? null : Arrays.stream(adaptersAnnotation.value())
+          .filter(adAnn -> adAnn.type() == finalType)
+          .findFirst().orElse(null);
+
+      Converter<?, ?> memberConverter = null;
+      if (adapterAnnotation != null) {
+        Class<? extends XmlAdapter> adapterCls = adapterAnnotation.value();
+        XmlAdapter<?, ?> adapter = ClassUtil.createInstance(adapterCls, true);
+
+        TypeFactory tf = soapstoneConfiguration.getObjectMapper().getTypeFactory();
+        JavaType adapterType = tf.constructType(adapter.getClass());
+        JavaType[] pt = tf.findTypeParameters(adapterType, XmlAdapter.class);
+        // Order of type parameters for Converter is reverse between serializer, deserializer,
+        // whereas JAXB just uses single ordering
+
+        memberConverter = new AdapterConverter(adapter, pt[1], pt[0], true);
+      }
+
+      if (memberConverter == null) {
```

Comment:
I think this second attempt to get the converter may be redundant, given what has been added to the `ParentAwareModelResolver`.

Reply:
done